db/collection.go

```diff
@@ -1221,6 +1202,38 @@ func (c *collection) saveFieldToMerkleCRDT(
 		)
 
 		return merkleCRDT.Set(ctx, bytes)
+	case client.PN_COUNTER:
```

Comment:
If you have a good suggestion on how to achieve this let me know. The mutator methods for each CRDT types will likely be different so I'm not sure how we could make this simpler.

Reply:
that's why I suggested a polymorphic approach (or something similar). Moreover, this `switch` case for the most part replicates another one from `merklecrdt.InstanceWithStore`. Why not to reuse it?
It could look something like this then:
```go
merkleCRDT, err := merklecrdt.InstanceWithStore(...)
// if err
merkleCRDT.Set(ctx, val) // this can be polymorphic. LWW will set it as is, whereas PNCounter will increment

Reply:
Because we don't want to use `Set` for all the CRDT types as asked by John in a conversation above. I'll see if I can do a workaround.

Reply:
Because each crdt is unique in the name, number, and type of arguments, it is likely impossible to create a perfectly clean abstraction over it. The mutation functions are specifically omitted from the ReplicatedData interface because there is no overlap. 

The only thing we could realistically do here would be to create a single mutation signature, called "Set" with a variadic `any` args, but then we're just moving this same complexity from this switch (in a single place) to all CRDT implementations, which seems like a huge loss.

There may be a cleaner solution, but I believe it can be broken out of this PR and tackled separately 👍